torch/distributed/rpc.py

```diff
@@ -156,17 +164,19 @@ def rpc(to, func, args=None, kwargs=None, async_call=False):
         >>> dist.init_rpc("worker1")
         >>> dist.join_rpc()
     """
+    assert (callable(func)), "function should be callable."
```

Comment:
Use of assert here is incorrect. An AssertionFailure means that we, the library authors, messed up. This is error checking in the public API; you should raise a `TypeError` here.